vlm/src/main/scala/geotrellis/contrib/vlm/SourceMetadata.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm
+
+trait SourceMetadata {
+  def base: Map[String, String]
+  def band(b: Int): Map[String, String]
+  def combine(self: SourceMetadata, bandCount: Int): SourceMetadata
```

Comment:
I'd suggest documenting the semantics of this... not clear to me what the end result would be, what `bandCount` is specifying, what happens if keys overlap etc. IOW, what are the laws :-p

Reply:
@metasim in addition I also don't like the names as they don't describe semantics; if you have any better suggestions you're welcome!

Reply:
@pomadchin I'd need to know what it does before suggesting a name!

Reply:
@metasim I just removed it